src/commands/handlers/registration.js

```diff
@@ -354,6 +356,29 @@ var handlers = {
     }
 };
 
+/**
+ * Only use the nick+password combo if an account has not been specifically given.
+ * If no account:{account,password} has been given, use the nick+password if available
+ */
+function getSaslAuth(handler) {
+    let options = handler.connection.options;
+    if (options.account && options.account.account) {
+        return {
+            account: options.account.account,
+            password: options.account.password || '',
+        };
+    } else if (options.account && options.password) {
```

Comment:
Perhaps we should return null if `options.account` object exists at all, not just passwords.